lib/constructs/interface.js

```diff
@@ -111,9 +111,21 @@ Interface.prototype.generateConstructor = function () {
     this.str += `
   iface.setup(this${passArgs});
 }\n`;
+  } else if (this.name === 'HTMLElement') {
+    this.str += `function ${this.name}() {
+  const { document, customElements } = global.window;
```

Comment:
Chrome will auto-detect the environment; I don't know how to do this in jsdom. I'm cheating and using `global.window` which would be set like `global.window = jsdomInstance.window`. This is really bad and I want to have it infer directly from jsdom.

This is required since you cannot call `new` on a subclassed `HTMLElement` that has not been registered with the active `CustomElementsRegistry`.